moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -258,15 +48,11 @@ const findDifferenceByFeatureProperty = (featureProperty, arrayOne, arrayTwo) =>
 export function useMapDrawTools(
```

Comment:
dang, this hook is a beast! i don't think it's worth tackling now, but just observing that all of the code in this hook is running on any map interaction (pan/zoom), including the <ComponentsDrawControl/> being re-constructed. i'm kind of surprised the map experience doesn't really drag. i don't have a good grasp on all the state dependencies, so maybe breaking into smaller chunks would be really difficult. in any case—out of scope.

Reply:
Yes, it certainly is. I would love to see this tamed and broken down into more pieces in the future -  I believe it is possible! Definitely see some of the patterns from the Data & Performance Hub helping that along. 👍